### PR TITLE
[MERGE] write into main

### PR DESCRIPTION
**Pull request by Valentin-Kaiser**

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -46,6 +46,12 @@ linters:
     - structcheck
     - wastedassign
 
+    # review later before merging to master
```
Comment (Plaenkler) [inline .golangci.yml:49]: You wont use those linters?

File: dbase/table.go [hunk 2]
```diff
@@ -48,19 +36,37 @@ type Column struct {
...
-	Convert     func(interface{}) interface{}
-	ExternalKey string
+// Table is a struct containing the table columns, modifications and the row pointer
+type Table struct {
+	// Columns defined in this table
+	columns []*Column
+	// Modification to change values or name of fields
+	mods []*Modification
+	// Internal row pointer, can be moved
+	rowPointer uint32
```
Comment (Plaenkler) [inline dbase/table.go:46]: Tables have a row limit? uint32 -> 64

**Review by Plaenkler — APPROVED** [review]

File: dbase/table.go (hunk 2 above)
Comment (Valentin-Kaiser) [reply dbase/table.go:46]: > # Table Header Record Structure 
> 
> 
> 
> Byte offset | Description
> -- | --
> 0 | File type: 0x01FoxBASE: 0x02FoxBASE+/Dbase III plus, no memo: 0x2FVisual FoxPro: 0x30Visual FoxPro, autoincrement enabled: 0x31Visual FoxPro, Varchar, Varbinary, or Blob-enabled: 0x42dBASE IV SQL table files, no memo: 0x62dBASE IV SQL system files, no memo: 0x82FoxBASE+/dBASE III PLUS, with memo: 0x8AdBASE IV with memo: 0xCAdBASE IV SQL table files, with memo: 0xF4FoxPro 2.x (or earlier) with memo: 0xFA
> 1 - 3 | Last update (YYMMDD)
> 4 – 7 | Number of records in file
> 
> 
> 

**The number of records in the file header is defined by Microsoft as uint32.**

File: .golangci.yml (hunk 1 above)
Comment (Valentin-Kaiser) [reply .golangci.yml:49]: Nope, I won't. I will still review it in the future. Therefore, the comment can remain

**Review by Plaenkler — APPROVED** [review]

File: dbase/interpreter.go [hunk 3]
```diff
@@ -90,21 +92,212 @@ func (dbf *DBF) DataToValue(raw []byte, column *Column) (interface{}, error) {
...
 		}
 		return f, nil
 	default:
 		return nil, fmt.Errorf("dbase-interpreter-datatovalue-9:FAILED:Unsupported column data type: %s", column.Type())
 	}
 }
+
+// Converts column data to the byte representation
+// For M values the data has to be written to the memo file
+func (dbf *DBF) valueToData(field *Field) ([]byte, error) {
```
Comment (HighQualityWaschbaer) [inline dbase/interpreter.go:116]: In my opinion, valueToData does not correctly describe what this function actually does. Maybe something like "valueToByteRepresentation" or "valueToByteArray" would be more descriptive.

**Review by HighQualityWaschbaer — CHANGES_REQUESTED** [review]

**Review by HighQualityWaschbaer — APPROVED** [review]

**Pull request merged**